src/conn/conn_prefetch.c

```diff
@@ -158,6 +158,34 @@ __wt_conn_prefetch_queue_push(WT_SESSION_IMPL *session, WT_REF *ref)
 
     conn = S2C(session);
 
+    /*
+     * We want to avoid the scenario of requesting pre-fetch on one particular ref many times (e.g
+     * when reading along a single page). We can identify this by checking if the previous pre-fetch
+     * was performed using the same home ref.
+     *
+     * In the event that we find this to be true, we perform pre-fetch for approximately the number
+     * of pages that were added to the queue (WT_PREFETCH_QUEUE_PER_TRIGGER). We then want to ensure
+     * that we will not pre-fetch from this ref for a while, and this is done by checking a counter.
+     *
+     * The counter variable prefetch_skipped_with_parent tracks the number of skips we have
+     * performed on a particular ref. If the number of skips surpasses the number of pages that have
+     * been queued for pre-fetch, we are okay to pre-fetch from this ref again. This condition will
+     * evaluate to false and the counter will be reset, effectively marking the ref as available to
+     * pre-fetch from.
+     */
+
+    if (session->pf.prefetch_prev_ref->page == ref->home) {
```

Comment:
I think the way this current condition is written, we will never reset the `session->pf.prefetch_skipped_with_parent` variable back to zero when we start pre-fetching from a different `ref`. Do we need to add another `session->pf.prefetch_skipped_with_parent = 0;` statement in an `else` block too?

Reply:
Yes you're right, thanks for pointing that out. I'll just move the `session->pf.prefetch_skipped_with_parent = 0;` statement out of the if/else block entirely.